src/language-server/engine/index.ts

```diff
@@ -27,42 +30,66 @@ export function noServiceError(service: string | undefined, endpoint?: string) {
-    private clientIdentity?: ClientIdentity,
+    private readonly engineKey: string,
+    baseURL: string = DefaultEngineConfig.endpoint,
+    private readonly clientIdentity?: ClientIdentity,
   ) {
-    super();
-    this.baseURL = engineEndpoint;
-  }
-
-  // XXX fix typings on base package
-  willSendRequest(request: any) {
-    if (!request.headers) request.headers = {};
-    request.headers["x-api-key"] = this.engineKey;
-    if (this.clientIdentity && this.clientIdentity.name) {
-      request.headers["apollo-client-name"] = this.clientIdentity.name;
-      request.headers["apollo-client-reference-id"] =
-        this.clientIdentity.referenceID;
-      request.headers["apollo-client-version"] = this.clientIdentity.version;
-      return;
-    }
-
-    // default values
-    request.headers["apollo-client-name"] = "Apollo Language Server";
-    request.headers["apollo-client-reference-id"] =
-      "146d29c0-912c-46d3-b686-920e52586be6";
-    request.headers["apollo-client-version"] =
-      require("../../package.json").version;
+    const link = ApolloLink.from([
+      onError(({ graphQLErrors, networkError, operation }) => {
+        const { result, response } = operation.getContext();
+        if (graphQLErrors) {
+          graphQLErrors.map((graphqlError) =>
+            console.error(`[GraphQL error]: ${graphqlError.message}`),
+          );
+        }
+
+        if (networkError) {
+          console.log(`[Network Error]: ${networkError}`);
+        }
+
+        if (response && response.status >= 400) {
+          console.log(`[Network Error] ${response.bodyText}`);
+        }
+      }),
+      setContext((_, request) => {
+        if (!request.headers) request.headers = {};
+        request.headers["x-api-key"] = this.engineKey;
+        if (this.clientIdentity && this.clientIdentity.name) {
+          request.headers["apollo-client-name"] = this.clientIdentity.name;
+          request.headers["apollo-client-reference-id"] =
+            this.clientIdentity.referenceID;
+          request.headers["apollo-client-version"] =
+            this.clientIdentity.version;
+        } else {
+          // default values
+          request.headers["apollo-client-name"] = "Apollo Language Server";
+          request.headers["apollo-client-reference-id"] =
+            "146d29c0-912c-46d3-b686-920e52586be6";
+          request.headers["apollo-client-version"] =
+            require("../../package.json").version;
+        }
+        return request;
+      }),
+      createHttpLink({ uri: baseURL }),
+    ]);
+
+    this.client = new ApolloClient({
+      link,
+      cache: new InMemoryCache(),
+    });
   }
 
   async loadSchemaTagsAndFieldLatencies(serviceID: string) {
-    const { data, errors } = await this.execute<SchemaTagsAndFieldStatsQuery>({
+    const { data, errors } = await this.client.query({
       query: SCHEMA_TAGS_AND_FIELD_STATS,
       variables: {
         id: serviceID,
       },
+      fetchPolicy: "no-cache",
```

Comment:
This is only executed when the config file is updated or the extension is loaded - it should probably not be cached. (It never was in the past)